### PR TITLE
PIM-10318: [Backport PIM-10259] Add ArPHP library and fix Arabic RTL in PDF Exports

### PR DESCRIPTION
**Pull request by guillaumefer-akeneo**
Backport of https://github.com/akeneo/pim-community-dev/pull/16269/files
See https://akeneo.atlassian.net/browse/PIM-10259

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -81,6 +81,7 @@
         "guzzlehttp/guzzle": "6.5.5",
         "imagine/imagine": "1.2.4",
         "justinrainbow/json-schema": "^5.2",
+        "khaled.alshamaa/ar-php": "^6.2",
```
Comment (nmarniesse) [inline composer.json:84]: Normally we can't add a library in a patch. But the GA is not available yet... What do you think @ahocquard ?  
Without this lib we can't fix the issue in 6.0

File: src/Akeneo/Pim/Enrichment/Bundle/PdfGeneration/Builder/DompdfBuilder.php [hunk 2]
```diff
@@ -24,12 +25,15 @@ class DompdfBuilder implements PdfBuilderInterface
      */
     protected $dompdf;
 
+    protected HtmlFormatter $arabicHtmlFormatter;
+
     private string $publicDir;
 
-    public function __construct(string $rootDir, $publicDir)
+    public function __construct(string $rootDir, $publicDir, HtmlFormatter $arabicHtmlFormatter)
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Bundle/PdfGeneration/Builder/DompdfBuilder.php:32]:
```suggestion
    public function __construct(string $rootDir, $publicDir, HtmlFormatter $arabicHtmlFormatter = null)
```

Adding an argument is a BC break. We need to set it optional

File: src/Akeneo/Pim/Enrichment/Bundle/PdfGeneration/Builder/DompdfBuilder.php [hunk 3]
```diff
@@ -24,12 +25,15 @@ class DompdfBuilder implements PdfBuilderInterface
      */
     protected $dompdf;
 
+    protected HtmlFormatter $arabicHtmlFormatter;
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Bundle/PdfGeneration/Builder/DompdfBuilder.php:28]:
```suggestion
    protected ?HtmlFormatter $arabicHtmlFormatter;
```

File: src/Akeneo/Pim/Enrichment/Bundle/PdfGeneration/Builder/DompdfBuilder.php [hunk 4]
```diff
@@ -57,6 +61,8 @@ protected function render($html)
             'chroot' => $this->publicDir
         ]);
         $this->dompdf = new Dompdf($options);
+
+        $html = $this->arabicHtmlFormatter->formatHtml($html);
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Bundle/PdfGeneration/Builder/DompdfBuilder.php:65]:
```suggestion
        if (null !== $this->arabicHtmlFormatter) {
            $html = $this->arabicHtmlFormatter->formatHtml($html);
        }
```

File: composer.json (hunk 1 above)
Comment (ahocquard) [reply composer.json:84]: If it's not GA yet, it's ok for me. Just precise it in the README.md to be sure.

File: src/Akeneo/Pim/Enrichment/Bundle/PdfGeneration/Builder/DompdfBuilder.php (hunk 2 above)
Comment (guillaumefer-akeneo) [reply src/Akeneo/Pim/Enrichment/Bundle/PdfGeneration/Builder/DompdfBuilder.php]: :+1: ok, it's noted, thx

**Pull request merged**